### PR TITLE
DSD-1507: Updating DS internal theming and DSProvider component

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Partially fixes JIRA ticket [DSD-1507](https://jira.nypl.org/browse/DSD-1507)

## This PR does the following:

- Adds `resetCSS` and `disableGlobalStyle` props to the `DSProvider`.
- Adds an internal `ChakraProvider` component in order to include fixes we want for conditionally disabling global styles and the reset CSS rules.
- This is a code change to the library that does NOT affect any consuming application. The reset CSS and global style rules will render by default as they currently do.
- This update is to support the NYPL Header app when in used in applications that have their own design system or base styles and the DS styles should not conflict. This also helps reduce the double CSS style rules coming from the Header app on existing DS apps.

## How has this been tested?

Locally there is no difference. But this was tested through a beta npm release version deployed in the Header app on https://nypl-header-fix3uj2y6-nypl.vercel.app/. 

To test: visit the Vercel preview and inspect the HTML and Body elements and make sure that there are no base styles set. This can be compared with the current header app.

`<body>` element rules in the production DS Header:
Screen Shot 2023-06-13 at 5 13 38 PM

`<body>` element rules in the Vercel Update:
Screen Shot 2023-06-13 at 5 14 04 PM

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 - Adds `"xxxsmall"` size (32px width) to the `Image` component.
 - Adds the `"connect"` section hues to the light and dark color palettes.
 - Adds the `"connect"` color variant to the `Breadcrumbs` component.
+- Adds internal `ChakraProvider` component to be able to conditionally include CSS styles through the `resetCSS` and `disableGlobalStyle` props.
+- Adds a `theme` prop to the `DSProvider` to merge in an additional theme file if necessary; this is to support to the `Header` and `Footer` components in the NYPL Header App.
```
Comment (bigfishdesign13) [inline CHANGELOG.md:17]: Do we have documentation on how to do this?  Or what the theme file would look like?

File: src/docs/Chakra.mdx [hunk 2]
```diff
@@ -81,29 +81,30 @@ const Application = (props) => {
   language="tsx"
 />
 
-### Chakra Components in Storybook
+### Disabling Global Styles
```
Comment (bigfishdesign13) [inline src/docs/Chakra.mdx]: I feel this should be a separate page under the `Development Guide`.

**Review by bigfishdesign13 — APPROVED** [review]
I think it looks good and it seems to work, but please see my comments/questions.

File: CHANGELOG.md (hunk 1 above)
Comment (EdwinGuzman) [reply CHANGELOG.md:17]: I didn't want to make this too much of a public feature, unless if you think it's okay to do?

File: src/docs/Chakra.mdx (hunk 2 above)
Comment (EdwinGuzman) [reply src/docs/Chakra.mdx]: Good idea.

File: CHANGELOG.md (hunk 1 above)
Comment (bigfishdesign13) [reply CHANGELOG.md:17]: Good point.  Let's hold off on adding explicit documentation for this.

File: src/components/DevelopmentGuide/DisablingGlobalStyles.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import { Meta, Source } from "@storybook/blocks";
+
+import Link from "../components/Link/Link";
+
+<Meta title="Disabling Global Styles" />
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/DisablingGlobalStyles.mdx]: This needs to be adjusted to group this page under `Development Guide`.
```
<Meta title="Development Guide/Disabling Global Styles" />
```

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.  You just need to adjust the title to get the page to sit properly in the sidebar.

**Pull request merged**